daemon/api_sideload_n_try_test.go

```diff
@@ -1093,3 +1093,30 @@ func (s *trySuite) TestTryChangeConflict(c *check.C) {
 	rspe := daemon.TrySnap(st, tryDir, snapstate.Flags{}).(*daemon.APIError)
 	c.Check(rspe.Kind, check.Equals, client.ErrorKindSnapChangeConflict)
 }
+
+func (s *sideloadSuite) TestSideloadSnapInvalidTransaction(c *check.C) {
+	s.daemon(c)
+
+	// try a multipart/form-data upload with missing "name"
+	content := "" +
+		"----hello--\r\n" +
+		"Content-Disposition: form-data; filename=\"x\"\r\n" +
+		"\r\n" +
+		"xyzzy\r\n" +
+		"----hello--\r\n"
+	content += "Content-Disposition: form-data; name=\"transaction\"\r\n" +
+		"\r\n" +
+		"xyz\r\n" +
+		"----hello--\r\n"
```

Comment:
Actually, I think this whole thing can be simplified a bit by removing the form part with the missing name
```suggestion
	content := "" +
		"----hello--\r\n" +
		"Content-Disposition: form-data; name=\"transaction\"\r\n" +
		"\r\n" +
		"xyz\r\n" +
		"----hello--\r\n"
```